api/openapi/model-registry.yaml

```
@@ -0,0 +1,1041 @@
+    BaseExecutionCreate:
+      allOf:
+        -
+          $ref: '#/components/schemas/BaseExecutionUpdate'
+        -
+          type: object
+        -
+          $ref: '#/components/schemas/BaseResourceCreate'
+    BaseExecutionUpdate:
+      type: object
+      allOf:
+        -
+          type: object
+          properties:
+            lastKnownState:
+              $ref: '#/components/schemas/ExecutionState'
+        -
+          $ref: '#/components/schemas/BaseResourceUpdate'
+    MetadataValue:
+      oneOf:
+        -
+          type: object
+          properties:
+            int_value:
+              format: int64
+              type: string
+        -
+          type: object
+          properties:
+            double_value:
+              format: double
+              type: number
+        -
+          type: object
+          properties:
+            string_value:
+              type: string
+        -
+          type: object
+          properties:
+            struct_value:
+              description: Base64 encoded bytes for struct value
+              type: string
+        -
+          type: object
+          properties:
+            type:
+              description: url describing proto value
+              type: string
+            proto_value:
+              description: Base64 encoded bytes for proto value
+              type: string
+        -
+          type: object
+          properties:
+            bool_value:
+              type: boolean
+      description: A value in properties.
+    BaseResource:
+      allOf:
+        -
+          $ref: '#/components/schemas/BaseResourceCreate'
+        -
+          type: object
+          properties:
+            id:
+              format: int64
+              description: Output only. The unique server generated id of the resource.
+              type: string
+              readOnly: true
+            createTimeSinceEpoch:
+              format: int64
+              description: Output only. Create time of the resource in millisecond since epoch.
+              type: string
+              readOnly: true
+            lastUpdateTimeSinceEpoch:
+              format: int64
+              description: |-
+                Output only. Last update time of the resource since epoch in millisecond
+                since epoch.
+              type: string
+              readOnly: true
+    BaseResourceCreate:
+      allOf:
+        -
+          $ref: '#/components/schemas/BaseResourceUpdate'
+        -
+          type: object
+          properties:
+            name:
+              description: |-
+                The client provided name of the artifact. This field is optional. If set,
+                it must be unique among all the artifacts of the same artifact type within
+                a database instance and cannot be changed once set.
+              type: string
+    BaseResourceUpdate:
+      type: object
+      properties:
+        customProperties:
```

```suggestion
        metadata:
```

open question, why not calling it directly metadata?

---

IMO, `custom properties` convey the intent more clearly as they are arbitrary data. `metadata` without a schema attached may be vague.

---

These are meant for user provided custom properties.